vignettes/sop.Rmd

```diff
@@ -490,185 +543,158 @@ compare_ranges <- c_spec(list(none, r1, r2), range = "common")
 
-```{r, fig.align="center", echo=FALSE}
-knitr::include_graphics("sop/matchplot.png")
-```
+Overlaying unknown spectra and the best matches can be extremely useful to identify peaks that don't fit to the reference library which may need further investigation. The example below shows great correspondence between the best match and the unknown spectrum. All major peaks are accounted for and the correct relative height. There are two small peaks in the unknown spectrum near 500 that are not accounted for which could be investigated further but we would call this a positive id to HDPE. 
 
-This plot is dynamically updated by selecting matches from the match table. The
-red spectrum is the spectrum that you selected from the reference library and
-the white spectrum is the spectrum that you are trying to identify. Whenever a
-new dataset is uploaded, the plot and data table in this tab will be updated.
-These plots can be saved as a .png by clicking the camera button at the top of
-the plot.
+```{r}
+plotly_spec(processed, filter_spec(test_lib, logic = matches[[1,"library_id"]]))
+```
 
-### Sharing Reference Data
+## Sharing Reference Data
 
 If you have reference data or AI models that you think would be useful for
 sharing with the spectroscopy community through OpenSpecy please contact the
-website administrator to discuss options for collaborating.
+package administrator to discuss options for collaborating.
+
+# Characterizing Particles
+
+Sometimes the spectroscopy task we want to perform is to identify particles in a spectral map. This is especially common for microplastic analysis where a spectral map is used to image a sample and spectral information is used to differentiate microplastic particles from nonplastic particles. In addition to the material id, one often wants to measure the shape and size of the particles. In a brute force technique, one could first identify every spectrum in the map, then use thresholding and image analysis to measure the particles. However, more often than not, particles are well separated on the image surface and background spectra is quite different from particle spectra and therefore we can use thresholding a priori to identify and measure the particles, then pass an exemplary spectrum for each particle to the identification routine. It is important to note here that this is at the bleeding edge of theory and technique so we may be updating these functions in the near future. 
 
-## Characterizing Particles
+## Brute Force
+
+```{r eval = F}
+data("test_lib")
+test_map <- read_any(read_extdata("CA_tiny_map.zip"))
 
-### Thresholding
+test_map_processed <- process_spec(test_map, conform_spec_args = list(range = test_lib$wavenumber, res = NULL))
 
-### Collapse Spectra
+identities <- match_spec(test_map_processed, test_lib, add_library_metadata = "sample_name", top_n = 1)[order(factor(object_id, levels = colnames(test_map$spectra)))]#Should integrate this sorting into match_spec, tried but couldn't get it. 
```

Comment:
What do you want to do here? Maybe I can add it to `match_spec()`?

Reply:
It's just ordering by `object_id`, isn't it?

Reply:
Yeah but needs to be the same order as the original data so that people can grab and go on the values in the output and related them to their spectra.